processing/src/main/java/io/druid/query/topn/PooledTopNAlgorithm.java

```diff
@@ -40,6 +45,19 @@
 public class PooledTopNAlgorithm
     extends BaseTopNAlgorithm<int[], BufferAggregator[], PooledTopNAlgorithm.PooledTopNParams>
 {
+  /** Non-final fields for testing, see TopNQueryRunnerTest */
+  @VisibleForTesting
+  static boolean specializeGeneric1AggPooledTopN =
+      !Boolean.getBoolean("dontSpecializeGeneric1AggPooledTopN");
+  @VisibleForTesting
+  static boolean specializeGeneric2AggPooledTopN =
+      !Boolean.getBoolean("dontSpecializeGeneric2AggPooledTopN");
```

Comment:
do you intend these to be user configurable, in that case they should be part of TopNQueryConfig really